### ISSUE TITLE
Comparison with mmm-mode

### ISSUE THREAD

**Issue by yantar92** [polymode/polymode#187]
I am wondering how the polymode is different from [mmm-mode](https://github.com/purcell/mmm-mode).
It would be great to add the comparison to README.

**Comment by vspinu** [comment]
I don't know mmm-mode and never used it so I cannot provide a comparison. I think the biggest difference is that with polymode you move between indirect buffers and as a result you have full 100% emacs mode when inside an inner chunk without much ado. With mmm-mode each time you move into a new chunk the whole set of variables (syntax-table, font-lock, all local variables specific to a mode) are re-installed in the current buffer.

> Later, the regions with different names can be fontified differently. I imagine that such kind of feature can be also utilized to group the regions for flycheck processing, which would be amazing for org source blocks, for example.

I am not sure what you mean here. How do you name regions? Each multi-mode has its own conventions to name chunks and that's not part of polymode functionality. Except of, maybe, allowing navigation to specific chunks but that's better be placed into Imenu. 

Naming chunks is not necessary for flycheck. It must work on all regions with the same inner-mode. I still have to look into how to make flycheck work in polymode buffers. Does mmm-mode allow flycheck?

**Comment by yantar92** [comment]
> ... Does mmm-mode allow flycheck?

Not really. However, the idea could be useful if one implements the flycheck functionality in polymode.
Let me explain.

Consider that flycheck is working in polymode. If I understand you correctly, you intend to run flycheck on all the regions with the same inner-mode. But it may be useful in some situations to run flycheck (or anything similar) on several groups of chunks independently.

I take an example from org-mode source blocks. 
In org-mode, you can tangle the source blocks to different files depending on the `:tangle` parameter. 
While we are in the org buffer, it means that there can be several groups of the code blocks, which belong to different files and should be checked (with flycheck) independently. 

If we are able to set names for individual chunks, it would be a convenient way to group the chunks in the situation above. Say, we can simply set the chunk names as their tangle file name.

**Comment by yantar92** [comment]
> I still have to look into how to make flycheck work in polymode buffers

Probably, it can be done by copying the chunks to templorary buffer. `replace-buffer-contents` and `narrow-to-region` might be useful.

**Comment by vspinu** [comment]
> it means that there can be several groups of the code blocks, which belong to different files and should be checked (with flycheck) independently.

I see. I don't think polymode should provide such grouping but if the underlying infrastructure (org, md, noweb) do provision for such groups then polymode better be understanding those indeed. 

> Probably, it can be done by copying the chunks to templorary buffer

Indeed, this seems to be the right way to go about it.
